krod/core/vector_store.py

```
@@ -235,43 +303,66 @@ def search_by_metadata(self, filter_dict: Dict[str, Any], limit: int = 10) -> Li
         if not query or not isinstance(query, str):
             raise ValueError("Query must be a non-empty string")
         
         if top_k < 1:
             raise ValueError("top_k must be a positive integer")
 
         try:
-            # Generate query embedding
-            query_embedding = self.embedding_model.encode(query).tolist()
+            # Check if collection exists and has points
+            try:
+                collection_info = self.client.get_collection(self.collection_name)
```

Pylint (3.3.7)

[error] Instance of 'VectorStore' has no 'client' member

(E1101)